authentication/builder.go

```diff
@@ -13,6 +13,9 @@ var (
 // Builder supports all of the possible Authentication values and feature toggles
 // required to build a working Config for Authentication purposes.
 type Builder struct {
+	// Context for passing to auth methods that accept one
+	Context context.Context
```

Comment:
(as above) can we move this onto the methods rather than holding this here, since this'll be different (with a different cancellation token) each time